File: kotlin/src/main/kotlin/com/joshy/hywrapper/model/skyblock/FiresalesResponse.kt
package com.joshy.hywrapper.model.skyblock

import com.joshy.hywrapper.model.HypixelResponse
import com.joshy.hywrapper.model.RateLimit
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlinx.serialization.Transient

/**
 * Lists all current news entires. These are viewable through Jerry on the island.
 *
 * @property success Whether the request was successful.
 * @property cause The cause of the error, if the request failed.
 * @property sales A list of all active firesales
 * */
@Serializable
data class FiresalesResponse(
    override val success: Boolean? = null,
    override val cause: String? = null,
    val sales: List<Sale>? = null,
) : HypixelResponse {
    @Transient
    override var rateLimit: RateLimit? = null
}

/**
 * Information about a firesale
 *
 * @property itemId The item ID of the item
 * @property start Timestamp of when the firesale begins
 * @property end Timestamp of when the firesale ends
 * @property amount Amount of items available for purchase
 * @property price Cost in gems for each item
 * */
@Serializable
data class Sale(
    @SerialName("item_id")
    val itemId: String? = null,
    val start: Long? = null,
    val end: Long? = null,
    val amount: Int? = null,
    val price: Int? = null,
)

File: typescript/src/models/skyblock/FiresalesResponse.ts
import { HypixelResponse } from "../../types";

/**
 * Lists all active SkyBlock fire sales.
 */
export interface FiresalesResponse extends HypixelResponse {
    /** A list of all active firesales */
    sales?: Sale[];
}

/**
 * Information about a firesale
 */
export interface Sale {
    /** The item ID of the item */
    item_id?: string;
    /** Timestamp of when the firesale begins */
    start?: number;
    /** Timestamp of when the firesale ends */
    end?: number;
    /** Amount of items available for purchase */
    amount?: number;
    /** Cost in gems for each item */
    price?: number;
}
